### ISSUE TITLE
[Router][DX] Relative resource references should resolve against the config file's location

### ISSUE THREAD

**Issue by backbone87** [symfony/symfony#24982]
When using relative route resource references, these references are resolved against the kernel root dir and not like all other config files against the file's location itself.
This is also mentioned in the ticket implementing relative route resource references: https://github.com/symfony/symfony/pull/21231#issuecomment-272418629

This is especially painful because using file relative references *sometimes* work:
```yml
app:
    resource: ../../src/Action/
    type:     annotation
```
This config works for me in one project, but in another project, i had to use:
```yml
app:
    resource: Action/
    type:     annotation
```
And i dont know why it works in one and not the other, both use the 3.4-BETA2 and are bundle less apps with kernel at `/src/Kernel.php` and config file at `/etc/_common/_routing.yml`

**Comment by nicolas-grekas** [comment]
It should, but unfortunately it would be a BC break to change this behavior.
We must find a continuous migration path if we want to fix this...

**Comment by backbone87** [comment]
One idea that comes to my mind is using the same loader mechanism for routes, as for the DI container and let a bundle pass it to the route loader somehow. this would be an additional way to load routes and not a replacement. That way we can change the way the default recipe of the framework bundle for a soft migration:
- "deprecate" `config/routes` and move them to `config/packages` and the current implementation of the Kernels `configureRoutes` (not the method itself, just the content)
- let the framework bundle somehow pass the routes configuration array (loaded via DI loader) to the route loader.

**Comment by nicolas-grekas** [comment]
Actually, this has been fixed in #25113

@backbone87 that's a totally different story, not sure about it.

Closing here, as the reported issue is gone.
